extension-manifest-v3/src/background/utils/startup-guard.js

```diff
@@ -0,0 +1,56 @@
+export default (starter, stopper) => {
```

Comment:
If you want to have the option to abort a "pending" promise, we can use the `AbortController` API, as it is already widely supported: https://caniuse.com/abortcontroller (including Safari, which is our main target currently).